docs/sources/variables/add-query-variable.md

```diff
@@ -0,0 +1,99 @@
++++
+title = "Add a query variable"
+type = "docs"
+[menu.docs]
+weight = 500
++++
+
+# Add a query variable
+
+Query variables allow you to write a data source query that can return a list of metric names, tag values, or keys. For example, a query variable might return a list of server names, sensor IDs, or data centers. The variable values change as they dynamically fetch options with a data source query.
+
+Query expressions can contain references to other variables and in effect create linked variables. Grafana detects this and automatically refreshes a variable when one of its linked variables change.
+
+## Query expressions
+
+Query expressions are different for each data source. For more information, refer to the documentation for your data source:
+
+- [Graphite templating queries]({{< relref "../features/datasources/graphite.md#templating" >}})
+- [Elasticsearch templating queries]({{< relref "../features/datasources/elasticsearch.md#templating" >}})
+- [InfluxDB templating queries]({{< relref "../features/datasources/influxdb.md#templating" >}})
+- [Prometheus templating queries]({{< relref "../features/datasources/prometheus.md#templating" >}})
+- [OpenTSDB templating queries]({{< relref "../features/datasources/opentsdb.md#templating" >}})
+- [AzureMonitor templating queries]({{< relref "../features/datasources/azuremonitor.md#templating-with-variables-for-the-azure-monitor-service" >}})
```

Comment:
Should we link templating sections for all data sources here or why only a few? I guess due to historic reason, so maybe something that can be fixed in a separate PR?